### PR TITLE
Integration testing bugs; tests; Pep8

### PR DESCRIPTION
**Pull request by oudeismetis**
related to 95 in franklin

### CODE REVIEW

File: franklin/builder/tests.py [hunk 1]
```diff
@@ -78,25 +80,50 @@ def test_get_deployable_event_tag(self):
...
+    def test_building_env(self, mock_post):
         """ Tests the model method that calls franklin-builder when an
         environment is ready to be deployed for a branch.
         """
+        # Mock the POST request to builder
+        response = mock.Mock()
+        response.status_code = 200
+        mock_post.return_value = response
+
+        self.assertEqual(self.env.status, Environment.REGISTERED)
```
Comment (oudeismetis) [inline franklin/builder/tests.py]: This test had the potential to call an actual api previously. Am now mocking the requests library endpoint to return a success. We will want to use this kind of mocking more in the future.

File: franklin/core/tests.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            "environment_id": 1
+            }
+
+    @mock.patch('core.helpers.requests.post', side_effect=ConnectionError)
+    def test_make_rest_post_call_conn_error(self, mock_post):
+        """ Tests make_rest_post_call when the call has an Connection exception.
+        """
+        response = make_rest_post_call(self.url, self.headers, self.body)
+        self.assertEqual(response.status_code, 500)
+
```
Comment (oudeismetis) [inline franklin/core/tests.py]: Now that I know how to mock the request library properly, I went ahead and created a bunch of tests to validate some of our helper functions.

Found a bug in the process.

File: franklin/core/views.py [hunk 3]
```diff
@@ -71,7 +74,32 @@ def post(self, request, *args, **kwargs):
...
+            type: string
+            description: All is well
+            required: true
+    responseMessages:
+        - code: 4XX
+          message: Check response message. API may be healthy, but unable to \
+                  call builder
+        - code: 5XX
+          message: Check response message. API may be healthy, but unable to \
+                  call builder
```
Comment (oudeismetis) [inline franklin/core/views.py]: forgot to fix this.

File: franklin/builder/models.py [hunk 4]
```diff
@@ -208,41 +204,30 @@ class Environment(models.Model):
...
-                    else:
-                        logger.error("Negative response from Builder")
-                else:
-                    logger.error('Builder responded without json')
+
+            response = make_rest_post_call(url, headers, body)
+            if not status.is_success(response.status_code):
+                logger.error("Negative response from Builder: %s",
+                             response.status_code)
+                self.status = self.FAILED
```
Comment (oudeismetis) [inline franklin/builder/models.py]: This refactor is now compatible with builder. Fixes that bug we saw where API would report a failure even though builder was succeeding.

**Comment by oudeismetis** [comment]
I called out the important pieces above. The rest of the changes are gentle refactors, tests, and PEP8 stuff.

**Comment by JulianGindi** [comment]
This PR is :fire: :fire: :fire: :fire: 

:+1: from me

**Comment by CorneliusIV** [comment]
This makes me happy :+1:

**Pull request closed without merging**